### PR TITLE
[SC-230] Add Fixed Price Oracle

### PR DESCRIPTION
**Pull request by hexonaut**
An oracle that is hard-coded to a particular price.

Also going to ignore coverage here for now.

### CODE REVIEW

File: test/FixedPriceOracle.t.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    FixedPriceOracle oracle;
+
+    function setUp() public {
+        oracle = new FixedPriceOracle(123e8);
+    }
+
+    function test_constructor() public {
+        assertEq(oracle.price(), 123e8);
+        assertEq(oracle.decimals(), 8);
+    }
```
Comment (lucas-manuel) [inline test/FixedPriceOracle.t.sol]:
```suggestion
    function test_constructor() public {
        assertEq(oracle.price(),    123e8);
        assertEq(oracle.decimals(), 8);
    }
```

File: test/FixedPriceOracle.t.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    function test_bad_price_values() public {
+        vm.expectRevert("FixedPriceOracle/invalid-price");
+        new FixedPriceOracle(0);
+
+        vm.expectRevert("FixedPriceOracle/invalid-price");
+        new FixedPriceOracle(-1);
+    }
+
```
Comment (lucas-manuel) [inline test/FixedPriceOracle.t.sol:28]: Add a test to make sure that oracle.latestAnswer() == oracle.price()

File: test/FixedPriceOracleIntegration.t.sol [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    function setUp() public {
+        vm.createSelectFork(getChain("mainnet").rpcUrl, 18_492_860);
+
+        aaveOracle = IAaveOracle(0x8105f69D9C41644c6A0803fDA7D03Aa70996cFD9);
+        dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+        admin = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
+
+        oracle = new FixedPriceOracle(1e8);
+    }
```
Comment (lucas-manuel) [inline test/FixedPriceOracleIntegration.t.sol]:
```suggestion
    IAaveOracle aaveOracle = IAaveOracle(0x8105f69D9C41644c6A0803fDA7D03Aa70996cFD9);

    address dai   = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
    address admin = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;

    FixedPriceOracle oracle;

    function setUp() public {
        vm.createSelectFork(getChain("mainnet").rpcUrl, 18_492_860);

        oracle = new FixedPriceOracle(1e8);
    }
```

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Some small changes, also need to update forge-std to get CI to pass

File: test/FixedPriceOracle.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/FixedPriceOracle.t.sol]: done

File: test/FixedPriceOracle.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/FixedPriceOracle.t.sol:28]: oops totally missed that. added

File: test/FixedPriceOracleIntegration.t.sol (hunk 3 above)
Comment (hexonaut) [reply test/FixedPriceOracleIntegration.t.sol]: done

File: test/FixedPriceOracleIntegration.t.sol [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    FixedPriceOracle oracle;
+
+    function setUp() public {
+        vm.createSelectFork(getChain("mainnet").rpcUrl, 18_492_860);
+
+        oracle = new FixedPriceOracle(1e8);
+    }
+
+    function test_replace_dai_oracle() public {
+        assertEq(aaveOracle.getAssetPrice(dai), 1.00001809e8);
```
Comment (barrutko) [inline test/FixedPriceOracleIntegration.t.sol:27]: I assume that this is the price given by the oracle at the block that the fork is pinned to. A small comment, description or extraction to a var with a meaningful name would be nice.

**Review by barrutko — COMMENTED** [review]
One super minor thing. Other than that LGTM.

File: test/FixedPriceOracleIntegration.t.sol (hunk 4 above)
Comment (hexonaut) [reply test/FixedPriceOracleIntegration.t.sol:27]: done

**Review by barrutko — APPROVED** [review]
LGTM

**Review by lucas-manuel — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**